core/src/processing/opengl/shaders/LineVert.glsl

```diff
@@ -34,54 +34,64 @@ attribute vec4 color;
+
+  vec4 p = projectionMatrix * posp;
+  vec4 q = projectionMatrix * posq;
+
+  // formula to convert from clip space (range -1..1) to screen space (range 0..[width or height])
+  // screen_p = (p.xy/p.w + <1,1>) * 0.5 * viewport.zw
+
+  // prevent division by W by transforming the tangent formula (div by 0 causes
+  // the line to disappear, see https://github.com/processing/processing/issues/5183)
+  // t = screen_q - screen_p
+  //
+  // tangent is normalized and we don't care which direction it points to (+-)
+  // t = +- normalize( screen_q - screen_p )
+  // t = +- normalize( (q.xy/q.w+<1,1>)*0.5*viewport.zw - (p.xy/p.w+<1,1>)*0.5*viewport.zw )
+  //
+  // extract common factor, <1,1> - <1,1> cancels out
+  // t = +- normalize( (q.xy/q.w - p.xy/p.w) * 0.5 * viewport.zw )
+  //
+  // convert to common divisor
+  // t = +- normalize( ((q.xy*p.w - p.xy*q.w) / (p.w*q.w)) * 0.5 * viewport.zw )
+  //
+  // remove the common scalar divisor/factor, not needed due to normalize and +-
+  // (keep viewport - can't remove because it has different components for x and y
+  //  and corrects for aspect ratio, see https://github.com/processing/processing/issues/5181)
+  // t = +- normalize( (q.xy*p.w - p.xy*q.w) * viewport.zw )
+
+  vec2 tangent = normalize((q.xy*p.w - p.xy*q.w) * viewport.zw);
+
+  // flip tangent to normal (it's already normalized)
+  vec2 normal = vec2(-tangent.y, tangent.x);
+
   float thickness = direction.w;
-  
-  if (thickness != 0.0) {  
-    vec4 posq = posp + modelviewMatrix * vec4(direction.xyz, 0);
-    posq.xyz = posq.xyz * scale;  
-    vec4 clipq = projectionMatrix * posq; 
-  
-    vec3 window_p = clipToWindow(clipp, viewport); 
-    vec3 window_q = clipToWindow(clipq, viewport); 
-    vec3 tangent = window_q - window_p;
-    
-    vec2 perp = normalize(vec2(-tangent.y, tangent.x));
-    vec2 offset = perp * thickness;
-
-    if (0 < perspective) {
-      // Perspective correction (lines will look thiner as they move away 
-      // from the view position).  
-      gl_Position.xy = clipp.xy + offset.xy;
-      gl_Position.zw = clipp.zw;
-    } else {
-      // No perspective correction.	
-      vec4 offsetp = windowToClipVector(offset, viewport, clipp.w);
-      gl_Position = clipp + offsetp;
-    }    
-  } else {
-    gl_Position = clipp;
-  }
-  
+  vec2 offset = normal * thickness;
+
+  // Perspective ---
+  // convert from world to clip by multiplying with projection scaling factor
+  // to get the right thickness (see https://github.com/processing/processing/issues/5182)
+  // invert Y, projections in Processing invert Y
+  vec2 perspScale = (projectionMatrix * vec4(1, -1, 0, 0)).xy;
+
+  // No Perspective ---
+  // multiply by W (to cancel out division by W later in the pipeline) and
+  // convert from screen to clip (derived from clip to screen above)
+  vec2 noPerspScale = p.w / (0.5 * viewport.zw);
+
+  gl_Position.xy = p.xy + offset.xy * mix(noPerspScale, perspScale, perspective > 0);
```

Comment:
Ah, the overload is in GLSL docs, but looks like only for GLSL 4.50. Try to leave there only `perspective` without anything, it is integer of 0 or 1 and should work on it's own.